pkg/apis/common/validate.go

```diff
@@ -23,6 +23,14 @@ func ValidateTLSConfig(tlsConfig *TLSConfig) error {
 		clientKeySet = true
 	}
 
+	if tlsConfig.InsecureSkipVerify {
```

Comment:
Is this right? When `InsecureSkipVerify` is true, we should let it pass and the rest does not matter right?

Reply:
yes! that's the point of it. It's in TLS documentation as well. Currently argo-events doesn't allow to connect with MSK with SASL,SSL. I can only find this way to make it provide that. 
https://github.com/argoproj/argo-events/issues/1277

Reply:
> Also, please run `make codegen`.

Do you know why is the codegen failing? it doens't fail on my side

Reply:
I mean, your current logic is when `InsecureSkipVerify` is true, it returns an error, this doesn't seem to be right.

Reply:
No, that's only in the Validation function because the validate function has an `error` return type. If I use the `Fmt.Sprint` I'd have to change to `string` and all the tests would have to change as well. Then we'd have this question again for the other validation conditions(fields) whether it's better for them to return error or string.

Reply:
Validation function returns an error means it will not go through. If the spec is valid, simply return nil.

Reply:
Yeah! Others tests have it as well. I was thinking of just having Assert.Nil() would be less significant. Done!